sr_data_visualization/scripts/sr_data_visualization/sr_data_visualizer_gui.py

```diff
@@ -0,0 +1,1095 @@
+from sr_utilities.hand_finder import HandFinder
+
+
+class SrDataVisualizer(Plugin):
+    def __init__(self, context):
+        self.init_complete = False
+        self.first_run = True
+        super(SrDataVisualizer, self).__init__(context)
+        self.setObjectName("SrDataVisualizer")
+        self._widget = QWidget()
+        ui_file = os.path.join(rospkg.RosPack().get_path('sr_data_visualization'), 'uis', 'hand-e_visualizer.ui')
+        loadUi(ui_file, self._widget)
+        if __name__ != "__main__":
+            context.add_widget(self._widget)
+        self._widget.setWindowTitle("Dexterous Hand Data Visualizer")
+
+        # Set white background color
+        p = self._widget.palette()
+        p.setColor(self._widget.backgroundRole(), Qt.white)
+        self._widget.setPalette(p)
+        self.tab_widget_main = self._widget.findChild(QTabWidget, "tabWidget_main")
+        self.tabWidget_motor_stats = self._widget.findChild(QTabWidget, "tabWidget_motor_stats")
+
+        # Change tabs background color
+        p = self.tab_widget_main.palette()
+        stylesheet = """
+            QTabWidget>QWidget>QWidget{background: white;}
+            """
+        p.setColor(self.tab_widget_main.backgroundRole(), Qt.white)
+        self.tab_widget_main.setStyleSheet(stylesheet)
+
+        self.tab_widget_main.currentChanged.connect(self.tab_change)
+        self.tabWidget_motor_stats.currentChanged.connect(self.tab_change_mstat)
+
+        motor_stat_keys_file = os.path.join(rospkg.RosPack().get_path('sr_data_visualization'), 'config',
+                                            'data_visualiser_motor_stat_keys.yaml')
+        parameters_file = os.path.join(rospkg.RosPack().get_path('sr_data_visualization'), 'config',
+                                       'data_visualiser_parameters.yaml')
+
+        self.font_offset = -3
+
+        self._widget.resizeEvent = self.on_resize_main
+
+        with open(motor_stat_keys_file, 'r') as stream:
+            try:
+                self.motor_stat_keys = yaml.load(stream)
+            except yaml.YAMLError as exc:
+                print(exc)
+
+        with open(parameters_file, 'r') as stream:
+            try:
+                data_loaded = yaml.load(stream)
+                self.initialize(data_loaded)
+            except yaml.YAMLError as exc:
+                print(exc)
+
+        self.include_tactile_plugin()
+
+        self.setup_radio_buttons()
+
+        self.tabWidget_motor_stats.setCurrentIndex(0)
+        self.tabWidget_motor_stats.setCurrentIndex(1)
+        self.tabWidget_motor_stats.setCurrentIndex(2)
+        self.tabWidget_motor_stats.setCurrentIndex(3)
+        self.tabWidget_motor_stats.setCurrentIndex(4)
+        self.tabWidget_motor_stats.setCurrentIndex(5)
+        self.tabWidget_motor_stats.setCurrentIndex(0)
+
+        self.change_graphs(all=True, type="motor_stat", ncol=1)
+
+        self.t0 = 0
+        self.init_complete = True
+        # TODO: refresh graphs on resize?
```

Comment:
is it still need to be done?

Reply:
err, I'm not sure. There's a bug that comes up sometimes where if the gui is moved between screens of different resolutions, sometimes data can get 'stuck' on the graph. I was thinking of adding a 'reset' button just in case this happens. Refreshing all graphs on resize is another option, but it would take a few seconds and might not solve the data getting 'stuck'. If it doesn't solve it, then I'd have to clear all the data from all the graphs on resize, which I don't really like.

Reply:
Removed